### PR TITLE
Add --dry-run

### PR DESCRIPTION
**Pull request by 404Wolf**
Add support for `--dry-run` for `vt checkout` `vt checkout -b` `vt push` and `vt pull`. 

Turns out there's a ton of complexity with all the git operations that we were able to hide by not dealing with tracking deltas. This is sort of amazing that it all works, I added a lot of tests.

We now keep track of deltas, and I modified doAtomically to allow not copying back to the target directory to do these "simulations." In an ideal world this could be done all in memory, but I think this is good enough for now.

### CODE REVIEW

File: src/vt/lib/FileState.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    return {
+      created: this.created,
+      modified: this.modified,
+      not_modified: this.not_modified,
+      deleted: this.deleted,
+    };
+  }
+
+  [Symbol.for("Deno.customInspect")](): string {
+    return JSON.stringify(this.toJSON(), null, 2);
```
Comment (404Wolf) [inline src/vt/lib/FileState.ts:201]: What deno shows when you `console.log` a FileState

**Comment by 404Wolf** [comment]
@maxmcd for review you might want to do `git diff 9ab0813..HEAD`. For some reason github isn't noticing that most of these commits were already merged (this is a branch from a branch, but since we're squashing I don't think this will break linearity).

**Comment by maxmcd** [comment]
@404Wolf yeah one of the complexities of squash merges. I would merge main into this branch and then push so that your merge base is pointing at the latest main. 

I'm guessing you branched off of your other wip branch and then when that was merged with squash and this one was pointed at main it no longer sees those commits on main.

**Comment by 404Wolf** [comment]
Further review:

Not ready for merge, checkout dirty checking needs a few logic fixes :/

**Comment by 404Wolf** [comment]
I'm going to merge this. There's probably some edge cases, this is a really really tricky feature to implement. I'll write tests soon for the CLI stuff.
